src/compiler/checker.ts

```diff
@@ -6170,6 +6174,37 @@ namespace ts {
             }
         }
 
+        function containsArgumentsReference(declaration: FunctionLikeDeclaration): boolean {
+            const links = getNodeLinks(declaration);
+            if (links.containsArgumentsReference === undefined) {
+                if (links.flags & NodeCheckFlags.CaptureArguments) {

```

Comment:
CaptureArguments is only set if we are checking the function, which is not always the case when we are resolving a call.